scripts/run-analyzers.py

```diff
@@ -17,15 +20,24 @@
 
 input_dir = args.input_dir or '.'
 
-print('Running the main analyzer...')
-MainAnalyzer(input_dir).main()
-print('Running the container analyzer...')
-ContainerAnalyzer(input_dir).main()
-print('Running the container RPM analyzer...')
-ContainerRPMAnalyzer(input_dir).main()
-print('Running the container Go analyzer...')
-ContainerGoAnalyzer(input_dir).main()
-print('Running the loose artifact analyzer...')
-LooseArtifactAnalyzer(input_dir).main()
-print('Running the post-analyzer...')
-PostAnalyzer(input_dir).main()
+analyzers = [MainAnalyzer, ContainerAnalyzer, ContainerRPMAnalyzer, ContainerGoAnalyzer,
+             LooseArtifactAnalyzer, PostAnalyzer]
+analyzer_failures = []
+
+for analyzer in analyzers:
+    print(f'Running {analyzer.__name__}...')
+    try:
+        analyzer(input_dir).main()
+    except AnalysisFailure as error:
+        # Don't continue if the main analyzer failed since other analyzers rely on it
+        if analyzer == MainAnalyzer:
+            print('MainAnalyzer failed with the following:\n{}'.format(error),
+                  file=sys.stderr)
+            sys.exit(3)
+
+        unsuccessful_analyzers.append(str(error))
```

Comment:
`unsuccessful_analyzers` is not declared anywhere.

Reply:
That was a mistake on my end after renaming the variable.